wallet-new/src/Cardano/Wallet/API/V1/Wallets.hs

```diff
@@ -15,8 +16,8 @@ type API =
                    :> PostCreated '[ValidJSON] (WalletResponse Wallet)
     :<|> "wallets" :> Summary "Returns all the available wallets."
                    :> WalletRequestParams
-                   :> FilterBy '["wallet_id", "balance"] Wallet
-                   :> SortBy   '["balance"] Wallet
+                   :> FilterBy '[P.WalletId, P.Balance] Wallet
+                   :> SortBy   '[P.Balance] Wallet
```

Comment:
@Martoon-00 My 2 cents, and maybe @parsonsmatt can disprove: I don't like too much these type aliases, because:

- Doesn't buy you any extra "type safety" (after all, those are already types)
- They introduce another layer of indirection you need to unpack (as in "Hey, what's this type is for? let me check")
- They don't allow you to see _at glance_ the final filter format. An advantage of having a `Symbol` here is that you have a WYSIWYG kind of interaction: the types you see there maps directly to filter in the final API, as if you were using them in the browser. Now we lost this ability 😭

Reply:
That's reasonable, but as far as I understand, arguments about extra layer of indirection and clearness at glance usually succumb when highly duplication arises, and that's exactly the same case. It doesn't really buy us type safety, but if I provide incorrect name I'll get some totally unclear error from GHC, which doesn't make development easier.
My impression is that in current machinery it's quite easy to make a mistake (inconsistent parameter names `FilterBy` and `SortBy`, inconsistent `toIndex` and `parseJSON`), plus there is no errors processing in it yet. Once we implement errors processing, and remove boilerplate (one parameter name mention for `SortBy`, one for `FilterBy`), then type aliases won't be needed much anymore.

Tl;dr I wish we have these names extracted to type aliases at least as temporal measure, till the moment we facilitate changes in all this stuff.

Reply:
These type aliases may not be useful if we go with the `"query_param" ?= ParamType` design in #2658.